ocaml/sdk-gen/component-test/jsonrpc-client/go/main_test.go

```diff
@@ -50,3 +58,537 @@ func TestLoginSuccess(t *testing.T) {
+		return
+	}
+
+	if result != expectedResult{
+		t.Log("The result returned not the same with expected!")
+		t.Fail()
+		return
+	}
+
+	err = session.Logout()
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+}
+
+func TestVMGetSnapshotTime(t *testing.T) {
+	session, err := getSession("test_id9")
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	var vmRef xenapi.VMRef = "vmref0"
+	var result time.Time
+	var expectedResult time.Time = time.Date(2024, time.April, 20, 12, 0, 0, 0, time.UTC)
+	result, err = xenapi.VM.GetSnapshotTime(session, vmRef)
+
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	if result != expectedResult{
+		t.Log("The result returned not the same with expected!")
+		t.Fail()
+		return
+	}
+
+	err = session.Logout()
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+}
+
+func TestVMQueryDataSource(t *testing.T) {
+	session, err := getSession("test_id10")
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+    
+	var vmRef xenapi.VMRef = "vmref0"
+	var result float64
+	expectedResult := 1.0
+
+	result, err = xenapi.VM.QueryDataSource(session, vmRef, "CPU0 usage")
+	if err != nil{
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	if result != expectedResult {
+		t.Log("The result returned not the same with expected!")
+		t.Fail()
+		return
+	}
+
+	err = session.Logout()
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+}
+
+func TestVMSetShadowMultiplierLive(t *testing.T) {
+	session, err := getSession("test_id11")
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	var vmRef xenapi.VMRef = "vmref0"
+	var multiplier float64 = 2.0
+
+	err = xenapi.VM.SetShadowMultiplierLive(session, vmRef, multiplier)
+	if err != nil{
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	err = session.Logout()
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+}
+
+func TestRpcError(t *testing.T) {
+	//Constuct an rpc client error
+	session, err := getSession("test_id12")
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	var vmRef xenapi.VMRef = "vmref0" 
+	_, err = xenapi.VM.GetRecord(session, vmRef)
+	if err == nil{
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	var rpcError = errors.New("call VM.get_record() on http://localhost:5000/jsonrpc status code: 200. Could not decode response body: json: cannot unmarshal string into Go struct field ResponseError.error.code of type int")
+	if err.Error() != rpcError.Error(){
+		t.Log("The expected error is not the same with the returned one!")
+		t.Fail()
+		return
+	}
+
+	err = session.Logout()
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	
+}
+
+func TestHttpError(t *testing.T) {
+	// Construct an error in RPC response that returned for method/parameter check not as expected
+	session, err := getSession("test_id13")
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	var vmRef xenapi.VMRef = "vmref0"
+	var httpError = errors.New("API error: code 500, message Rpc server failed to handle the client request!")
```

Comment:
I think we can change the https://github.com/xapi-project/xen-api/blob/feature/go_sdk/ocaml/sdk-gen/component-test/jsonrpc-server/server.py#L56 to be like:
```
 "data": test_id,
``` 
So we can include the "data" in error string. and don't need to use "strings.Contains".